docs/formats/raw.md

```diff
@@ -0,0 +1,98 @@
+# Raw Format
+
+:::note
+This page describes configuration options specific to the raw data format.
+See [top-level connector documentation](/connectors/) for general information
+about configuring input and output connectors.
+:::
+
+The **raw** data format can be used to configure an input connector to pass
+raw, unparsed, data buffers from the [transport connector](/connectors/sources) to the SQL program.
+Normally the byte stream received from the transport connector is parsed using the specified data [format](/formats),
+e.g., [JSON](/formats/json) or [Avro](/formats/avro). The parser converts the raw byte stream into a stream
+of SQL table records.  In some cases it is preferable to pass the raw bytes to the SQL program unmodified and
+perform all parsing in SQL. In particular this is useful for:
+
+* **Custom data formats:** If the required data format is not supported by built-in parsers, raw data can be
+  ingested into a SQL table and processed using a [UDF](/sql/udf).
+
+* **Parallelized parsing**: Parsing large volumes of incoming data can become a performance bottleneck.
+  Offloading parsing to the SQL program enables parallel execution, leveraging the multithreaded Feldera
+  SQL runtime.
+
+The raw format is currently only supported for input connectors.
+
+Note that the raw format supports only inserting records into the table and does not support deletions.
+This limitation exists because the raw data stream lacks the metadata to differentiate between inserts and deletes.
+
+## Configuring raw input connector
+
+In order to ingest raw data:
+
+1. Declare a table with a single column of type `VARBINARY [NOT NUL]` or `VARCHAR [NOT NULL]`.  In the latter case, the connector
+   will parse input data as a UTF-8 string and will fail to parse chunks that are not valid UTF-8.
+
+2. Configure an input connector with the format name `raw`.  The connector currently supports a single configuration
+   option, `mode`, which can be set to:
+   * `blob` (default) - ingest the entire data chunk received from the transport connector as a single SQL row.
+     For message-oriented transports, such as Kafka or Pub/Sub, an input chunk corresponds to a message.
+     For file-based transports, e.g., the [URL](/connectors/sources/http-get) connector or
+     the [S3](/connectors/sources/s3) connector, a chunk represents an entire file or object.
+   * `lines` - split the input byte stream on the new line character (`\n`) and ingest each line as a separate SQL row.
+
+:::note
+
+When using the `blob` mode with a file-oriented transport connector such as S3, the entire file or object is ingested
```

Comment:
is there an option to read multiple binary files each as a row?

Reply:
Not as part of the format. This is up to the transport connector. The S3 connector for example will read multiple files; each will be ingested as a separate row.